analyzers/src/SonarAnalyzer.Common/Extensions/CompilationExtensions.cs

```diff
@@ -35,4 +35,35 @@ internal static class CompilationExtensions
 
     public static bool References(this Compilation compilation, KnownAssembly assembly)
         => assembly.IsReferencedBy(compilation);
+
+    public static bool IsMemberAvailable(this Compilation compilation, KnownType type, string memberName, KnownType memberType = null, KnownType[] parameterTypes = null)
```

Comment:
I don't like the method signature.
* memberType is one of many tests you want to do as a caller on the member (e.g. IsStatic, IsOverride, etc.)
* parameterTypes is ignored for properties even though they do have [parameters](https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.ipropertysymbol.parameters?view=roslyn-dotnet-4.7.0).
* parameterTypes should be a params array but memberType is already optional so this is not possible

What about this instead?
```cs
public static bool IsMemberAvailable(this Compilation compilation, KnownType type, string memberName, Func<ISymbol, bool> memberCheck = null)
{
    // ...
    return memberCheck == null ? memberSymbols.Any() : memberSymbols.Any(memberCheck);
}
```

Reply:
You can combine this with the `KnownMethods`  functionality.